include/measurement_kit/net/connection.hpp

```diff
@@ -32,20 +34,18 @@ namespace net {
 using namespace measurement_kit::common;
 using namespace measurement_kit::dns;
 
-class ConnectionState {
-
+class Connection : public Dumb {
+  private:
     Bufferevent bev;
     dns::Request dns_request;
     unsigned int connecting = 0;
-    char *address = NULL;
-    char *port = NULL;
-    StringVector *addrlist = NULL;
-    char *family = NULL;
-    StringVector *pflist = NULL;
+    std::string address;
+    std::string port;
```

Comment:
At that point in the code it could be an integer as well as a string. In fact, later on the port value is passed to an overloaded function that could either receive the port as an integer or as a string.

Historically here there is a string because libneubot was designed also to be called from Python `ctypes` and moving checks in C++ simplified the glue code between Python and C++.